core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/FileAsyncRequestBody.java

```diff
@@ -80,17 +83,23 @@ public String contentType() {
 
     @Override
     public void subscribe(Subscriber<? super ByteBuffer> s) {
+        AsynchronousFileChannel channel = null;
         try {
-            AsynchronousFileChannel channel = openInputChannel(this.path);
+            channel = openInputChannel(this.path);
 
             // We need to synchronize here because the subscriber could call
             // request() from within onSubscribe which would potentially
             // trigger onNext before onSubscribe is finished.
-            Subscription subscription = new FileSubscription(channel, s, chunkSizeInBytes);
+            //
+            // Note: size() can throw IOE here
```

Comment:
Under what circumstance can it throw IOException?

Reply:
The [javadoc](https://docs.oracle.com/javase/7/docs/api/java/nio/channels/AsynchronousFileChannel.html#size()) doesn't say:

> If some other I/O error occurs